daml/Marketplace/ExchangeParticipant.daml

```diff
@@ -67,48 +69,78 @@ template ExchangeParticipant
           let requiredDepositAmount = if isBid then qty * price else qty * (binaryOption.maxPrice - price)
           assertMsg ("deposit should be for at least " <> show requiredDepositAmount <> " " <> deposit.asset.id.label)
             $ (deposit.asset.quantity >= requiredDepositAmount)
-          if requiredDepositAmount < deposit.asset.quantity then do
-            [splitCid, restCid] <- exercise depositCid AssetDeposit_Split with quantities = [requiredDepositAmount]
-            let order = Order with status = "New", orderId = -1, pair = binOption, depositCid = splitCid, ..
-            create OrderRequest with ..
-          else do
-            let order = Order with status = "New", orderId = -1, pair = binOption, ..
-            create OrderRequest with ..
-
-      nonconsuming ExchangeParticipant_PlaceBid : ContractId OrderRequest
+
+          orderDepositCid <- if requiredDepositAmount < deposit.asset.quantity
+                             then head <$> exercise depositCid AssetDeposit_Split with quantities = [requiredDepositAmount]
+                             else return depositCid
+          orderDeposit <- fetch orderDepositCid
+
+          debitRequestCid <- create DepositDebitRequest with
+              owner = exchParticipant
+              ownerAccount = orderDeposit.account
+              depositCid = orderDepositCid
+
+          let order = Order with
+                status = "New", orderId = -1, pair = binOption
+                asset = orderDeposit.asset, account = orderDeposit.account, ..
+          orderRequestId <- create OrderRequest with ..
+          return (orderRequestId, debitRequestCid)
```

Comment:
Could the `ExchangeParticipant_MakeOrder` choice be reused here as well?

Reply:
Indeed it can, good call. I had to leave the two asserts that were in `ExchangeParticipant_MakeOrder` since the `ExchangeParticipant_PlaceBinaryOptionOrder` performs an `AssetDeposit_Split` before exercising `ExchangeParticipant_MakeOrder` (which you wouldn't want to do if it fails the asserts), but it still DRYs it up a bit 👍